pkg/services/cloudmigration/api/api.go

```diff
@@ -32,25 +34,104 @@ func RegisterApi(
 }
 
 // RegisterAPIEndpoints Registers Endpoints on Grafana Router
-func (api *MigrationAPI) registerEndpoints() {
-	api.routeRegister.Group("/api/cloudmigrations", func(apiRoute routing.RouteRegister) {
-		apiRoute.Post(
-			"/migrate_datasources",
-			routing.Wrap(api.MigrateDatasources),
-		)
+func (cma *CloudMigrationAPI) registerEndpoints() {
+	cma.routeRegister.Group("/api/cloudmigration", func(cloudMigrationRoute routing.RouteRegister) {
+		// authentication
+		cloudMigrationRoute.Get("/authentication", routing.Wrap(cma.Authenticate))
```

Comment:
We want per migration authentication so that we can configure migration against multiple stacks in the future.

Reply:
you mean `/:stack/authentication`